sim/battle.ts

```diff
@@ -279,19 +279,16 @@ export class Battle extends Dex.ModdedDex {
 	}
 
 	clearWeather() {
-		if (!this.weather) {
-			return false;
-		}
-		const oldstatus = this.getWeather();
-		this.singleEvent('End', oldstatus, this.weatherData, this);
+		if (!this.weather) return false;
+		const prevWeather = this.getWeather();
+		this.singleEvent('End', prevWeather, this.weatherData, this);
 		this.weather = '';
 		this.weatherData = {id: ''};
 		return true;
 	}
 
 	effectiveWeather() {
-		if (this.suppressingWeather()) return '';
-		return this.weather;
+		return this.suppressingWeather() ? '' : this.weather;
 	}
```

Comment:
It may be just me, but I think ``return default`` is better than a hard to read ternary?

Reply:
Entirely fair, but unfortunately completely subjective. We don't really have any rules around when we return a ternary and when we do the double `return`, and my JS background is in writing for Web circa 2010 where code size is king, which is probably why I prefer the former (also because I write C/C++/Java for my day job).

In a similar vein - when do we omit the braces from an `if` (or even `for` loops sometimes :S)? The answer is once again murky.

I've tried to use the rule that if the ternary would be long or complex then I would use the double `return` - in this specific case the ternary is pretty short and I don't think its necessarily hard to read if you're familiar with the ternary operator? But I could be very mistaken.

Regarding `if` - if the statement is short and an early `return`/`continue`/`break` then no braces seems like a no brainer. If its a variable assignment (`if (!foo) foo = ...`) or a method call (`if (foo) bar()`) then I'm less sure its always the right thing to do (and I'm really not a big fan for braceless loops except in rare cases), but theres no standard in the codebase. Do you and @Zarel have guidelines/rules of thumb you follow? I don't really have a huge preference here, I'm used to automatic formatters which insert all the braces everywhere regardless.

Reply:
As far as I can tell, the codebase follows the pattern that if one of the code paths is falsey or an error, it should early return, thus making the semantic split more explicit.

Reply:
For sure, but that's not quite what I'm talking about here. When does the early return/throw use braces and when doesn't it? And there are oodles of places where we're not returning/throwing and still don't use braces.

FWIW - I reverted back to the `return default` approach here.

Reply:
Unfortunately, that's... quite arbitrary... But you may notice that with/without braces is most often defined in a per-file basis, according to who the author was...

Reply:
I'm fine with either approach here. This is one of those style issues I've never bothered nitpicking since I don't have a strong opinion.